src/main/java/org/scijava/io/handle/FileHandle.java

```diff
@@ -101,199 +101,199 @@ public Date lastModified() {
 
 	@Override
 	public long offset() throws IOException {
-		return raf().getFilePointer();
+		return exists() ? writeRaf().getFilePointer() : 0;
 	}
 
 	@Override
 	public long length() throws IOException {
-		return exists() ? raf().length() : -1;
+		return exists() ? writeRaf().length() : -1;
```

Comment:
Conceptually, this should be `readRaf()` since querying the length is a "read" operation. (I realize that in practice, it won't matter due to the `exists()` check beforehand, but the code looks strange/wrong with `writeRaf()` here.)